app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java

```diff
@@ -1,17 +1,49 @@
 package com.example.parkingsystemjava.mvp.presenter;
 
+import com.example.parkingsystemjava.entity.Reservation;
+import com.example.parkingsystemjava.listener.ListenerDateTime;
 import com.example.parkingsystemjava.mvp.contract.ParkingReservationContract;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
 
 public class ParkingReservationPresenter implements ParkingReservationContract.ReservationActivityPresenter {
 
     private ParkingReservationContract.ReservationActivityView view;
+    private ParkingReservationContract.ReservationActivityModel model;
 
-    public ParkingReservationPresenter(ParkingReservationContract.ReservationActivityView view) {
+    public ParkingReservationPresenter(ParkingReservationContract.ReservationActivityModel model, ParkingReservationContract.ReservationActivityView view) {
+        this.model = model;
         this.view = view;
     }
 
     @Override
-    public void showDatePicker() {
-        view.showEntryDatePicker();
+    public void showDatePicker(ListenerDateTime listenerDateTime) {
+        view.showDatePicker(listenerDateTime);
+    }
+
+    @Override
+    public void addReservation(Reservation reservation, int parkingLot) {
+        model.addReservation(reservation, parkingLot);
+        view.showConfirmationMessage(parkingLot);
+    }
+
+    @Override
+    public void setEntryExitDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showDateSelected(stringDate);
+    }
+
+    private String getStringFromDate(Calendar calendar) {
+        Date date = calendar.getTime();
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("hh: mm a dd-MMM-yyyy");
+        String inActiveDate = null;
+        try {
+            inActiveDate = simpleDateFormat.format(date);
+            System.out.println(inActiveDate);
```

Comment:
this its not neccessary, insteand off use Log library to print in the console a message

Reply:
done